### PR TITLE
177160902/fetch PlotTypes and PlotOptions from db

### PR DESCRIPTION
**Pull request by upendrakumbham**
This takes care of fetching `PlotTypes` and corresponding `Options ` from the DB based on experiment accession.

Sample expected JSON response is here: 
`
{
        "tsne": [{ "perplexity": 10 }, { "perplexity": 20 } ...],
       "umap": [{ "n_neighbours": 3 }, ... ]|
      }
`

Endpoint to see results: "/experiments/{experimentAccession}" from the `ExperimentController.java`

### CODE REVIEW

**Comment by lingyun1010** [comment]
Currently, the front-end data structure is like
 ``` 
 plotTypeDropdown: [
    {
      plotType: `UMAP`,
      plotOptions: [5, 10, 15, 20, 25, 50, 100]
    },
    {
      plotType: `tSNE`,
      plotOptions: [1, 5 ,10, 15, 20, 25, 30, 35, 40, 45, 50]
    }
  ]
```
We will work on the [pivotal story](https://www.pivotaltracker.com/story/show/177741428) anyway, but if this format will help you to simplify your work.

**Comment by upendrakumbham** [comment]
> Currently, the front-end required data structure is like
> 
> ```
> plotTypeDropdown: [
>    {
>      plotType: `UMAP`,
>      plotOptions: [5, 10, 15, 20, 25, 50, 100]
>    },
>    {
>      plotType: `tSNE`,
>      plotOptions: [1, 5 ,10, 15, 20, 25, 30, 35, 40, 45, 50]
>    }
>  ]
> ```
> 
> We will work on the [pivotal story](https://www.pivotaltracker.com/story/show/177741428) anyway, but if this format will help you to simplify your work.

Thanks, @lingyun1010,  PR responsible to fetch prototypes and it's corresponding options from the database. We may need to process this response further to suit our front end as per group discussion feedback.

**Review by lingyun1010 — APPROVED** [review]
All good apart from a tiny comment.

File: src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java [hunk 1]
```diff
@@ -108,4 +111,35 @@ public Integer fetchNumberOfCellsByExperimentAccession(String experimentAccessio
                 namedParameters,
                 Integer.class);
     }
+
+    private static final String SELECT_DISTINCT_T_SNE_PLOT_TYPES_AND_OPTIONS =
+            "SELECT DISTINCT method,option FROM scxa_coords AS coords " +
+                    ",jsonb_array_elements(coords.parameterisation) option " +
```
[the remark this answers is not in the document]
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java:117]: I move the comma to the above statement as we need this to separate two statements.
like this: 
```
FROM scxa_coords AS coords, " +                    
"jsonb_array_elements(coords.parameterisation) option
```

[the remark this answers is not in the document]
Comment (upendrakumbham) [reply src/main/java/uk/ac/ebi/atlas/experimentpage/tsneplot/TSnePlotDao.java]: Done

**Comment by alfonsomunozpomer** [comment]
Superseded by https://github.com/ebi-gene-expression-group/atlas-web-single-cell/pull/168

**Pull request closed without merging**